pilot/platform/kube/controller.go

```diff
@@ -326,20 +329,41 @@ func (c *Controller) Instances(hostname string, ports []string,
 					}
 				}
 			}
+			if len(out) == 0 {
+				err := fmt.Errorf("No data found for the query")
+				return out, err
+			}
 			return out, nil
 		}
 	}
 	return nil, nil
 }
 
 // HostInstances implements a service catalog operation
-func (c *Controller) HostInstances(addrs map[string]bool) ([]*model.ServiceInstance, error) {
+func (c *Controller) HostInstances(addrs map[string]interface{}) ([]*model.ServiceInstance, error) {
 	var out []*model.ServiceInstance
+	for _, val := range addrs {
+		svcNode, _ := val.(*proxy.Node)
+		_, namespace, err := parseNodeID(svcNode.ID)
+		if err != nil {
+			return out, err
+		}
+		namespace1, err := parseNodeDomain(svcNode.Domain)
+		if err != nil {
+			return out, err
+		}
+		if namespace != namespace1 {
+			err = fmt.Errorf("namespace in nodeID %q must be equal to that in domain %q", svcNode.ID, svcNode.Domain)
+			return out, err
```

Comment:
Suggestion: introduce a new model.parseProxyID function which by default returns the same string or whatever. Override in each platform. And move this code into the implementation under that function

Reply:
Sounds like a good idea.

Reply:
Can you use the term `workload ID`? Since that's sidecar specific, it's the same thing as proxy ID, but makes it clear that it's a property of the service registry.